### ISSUE TITLE
Cleaning up

### ISSUE THREAD

**Issue by PangbornIdentity** [AshesOfOwls/jquery.shapeshift#14]
More of a question, but , I was wondering for dynamic creation and destruction,
do we need to call a delete or destroy? From working with the plugin I'm guessing JQuery Widget Factory isnt being used. any plans for this?
cheers,
Rich

**Comment by AshesOfOwls** [comment]
Are you asking whether or not there will be a way to create or destroy elements from the grid?

If so, this is something that won't be implemented into shapeshift. The reason behind this is twofold. The first is cleanliness of code, I would prefer shapeshift to contain the minimum amount of code possible to achieve the grid result. Once more code is added in to create new features it starts becoming a mess and every other feature suffers because of that.

The second reason is because shapeshift already compensates for insertions or deletions without having to call a special shapeshift function. Every time that the arrange event is called on a container, it will get the currently visible child elements and put them into place. If you wanted to delete one or insert a new element, you can do that with your own javascript and then trigger the arrange event on the container by doing this:

$(container).trigger("ss-event-arrange")

And all the elements will rearrange accordingly. This means if you want to set up a system where it filters items but doesn't delete them, just change the attribute display: none; on the child element you want to hide, trigger ss-event-arrange, and you have filtering all set up. 

So in conclusion the way to think about shapeshift is that it only arranges elements - any deletion or insertion is just done manually before the arrangement.

Hope that helps.

**Comment by PangbornIdentity** [comment]
Thanks for the detailed response.   I have dynamic insertions and deletions working on elements within the grid, and I am using the trigger as described.  I was wondering for garbage collection, if I should be calling some 'destroy' method on the items before removing them.  I'm also using dynamic containers, so I was wondering if there was a way  to destroy the shapeshift containers before removing them from the screen.
ie http://api.jqueryui.com/jQuery.widget/#method-destroy 

cheers,
Rich

**Comment by AshesOfOwls** [comment]
Gotcha, at the moment there is no ability to do that, but seeing as though it will be important to remove all the bindings and such I will implement it in our next release. It should be up sometime over this weekend.

**Comment by AshesOfOwls** [comment]
Sorry for the wait on this, I am currently in the process of getting it working.

**Comment by PangbornIdentity** [comment]
No problem, looking forward to the next release

**Comment by AshesOfOwls** [comment]
Version 2.0 is getting released tonight which will have a destroy event that you can trigger on the container element. I will comment here again when 2.0 is up.

**Comment by AshesOfOwls** [comment]
Pushing up soon, closing
